src/CMakeLists.txt

```diff
@@ -20,10 +20,23 @@ option(BUILD_MODULETEST "Build the moduletest tool" ON)
 option(BUILD_SAMPLES "Build samples" OFF)
 option(COVERAGE "Enable code coverage" OFF)
 
-add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:-Wall;-Wextra;-Wunused;-Werror;-Wformat;-Wformat-security;-Wreorder;-Wno-nonnull;-Wno-unused-result;-Wunused-macros;-Wunused-const-variable=2>")
+add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:-Wall;-Wextra;-Wunused;-Werror;-Wformat;-Wformat-security;-Wreorder;-Wno-nonnull;-Wno-unused-result;-Wunused-macros;>")
 
 if (CMAKE_COMPILER_IS_GNUCC)
     add_compile_options("-Wno-psabi;-fPIC")
+
+    # GCC 4.8 - Only build for NRP support
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9")
+        message(STATUS "GCC 4.8.x detected! Building for NRP support only.")
+        set(BUILD_MODULES OFF)
+        set(BUILD_PLATFORM OFF)
+        set(BUILD_TESTS OFF)
+        set(BUILD_ADAPTERS OFF)
+    endif()
+
+    if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.3.0")
+        add_compile_options("-unused-const-variable=2")
+    endif()
 endif()

```

Comment:
This is not enough. Do a diff against the one in the 14 branch as we discussed. More than half of this CMakeList.txt needs to be excluded from there (all that deals with package, install, etc)